### PR TITLE
[improve][meta] Improve fault tolerance of blocking calls by supporting timeout

### PR DESCRIPTION
**Pull request by mattisonchao**
### Motivation

We've got a lot of blocking calls without a timeout in the BookKeeper metadata plugin. In a bad situation, it will trigger some deadlocks. 

### Modifications

- Improve the fault tolerance of blocking calls by supporting timeout.

### Verifying this change

- [x] Make sure that the change passes the CI checks.

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete`

### CODE REVIEW

**Review by hangc0276 — APPROVED** [review]

**Comment by hangc0276** [comment]
@horizonzy Please help take a look, thanks.

**Review by poorbarcode — APPROVED** [review]

**Review by coderzc — APPROVED** [review]

**Review by horizonzy — COMMENTED** [review]
https://github.com/apache/pulsar/blob/d06cda6cd8a58b8a7e0678183f05c08059ddb9b2/pulsar-metadata/src/main/java/org/apache/pulsar/metadata/bookkeeper/PulsarLedgerUnderreplicationManager.java#L448

Here should also be catch

File: pulsar-metadata/src/main/java/org/apache/pulsar/metadata/bookkeeper/PulsarLedgerUnderreplicationManager.java [hunk 1]
```diff
@@ -392,12 +397,13 @@ public void markLedgerReplicated(long ledgerId) throws ReplicationException.Unav
...
             Lock l = heldLocks.get(ledgerId);
             if (l != null) {
-                store.delete(getUrLedgerPath(ledgerId), Optional.of(l.getLedgerNodeVersion())).get();
+                store.delete(getUrLedgerPath(ledgerId), Optional.of(l.getLedgerNodeVersion()))
+                        .get(BLOCKING_CALL_TIMEOUT, MILLISECONDS);
             }
-        } catch (ExecutionException ee) {
-            if (ee.getCause() instanceof MetadataStoreException.NotFoundException) {
+        } catch (ExecutionException | TimeoutException ee) {
+            if (ee.getCause() != null && ee.getCause() instanceof MetadataStoreException.NotFoundException) {
```
[the remark this answers is not in the document]
Comment (coderzc) [reply pulsar-metadata/src/main/java/org/apache/pulsar/metadata/bookkeeper/PulsarLedgerUnderreplicationManager.java:404]: +1

[the remark this answers is not in the document]
Comment (mattisonchao) [reply pulsar-metadata/src/main/java/org/apache/pulsar/metadata/bookkeeper/PulsarLedgerUnderreplicationManager.java]: fixed

**Review by Technoboy- — APPROVED** [review]

**Review by horizonzy — APPROVED** [review]
LGTM

**Pull request merged**